dev/devicelab/README.md

```diff
@@ -211,31 +203,36 @@ A task runs in its own standalone Dart VM and reports results via Dart VM
 service protocol. This ensures that tasks do not interfere with each other and
 lets the CI system time out and clean up tasks that get stuck.
 
-## Adding tests to the CI environment
+## Adding tests to continuous integration
 
-The `manifest.yaml` file describes a subset of tests we run in the CI. To add
-your test edit `manifest.yaml` and add the following in the "tasks" dictionary:
-
-```
-  {NAME_OF_TEST}:
-    description: {DESCRIPTION}
-    stage: {STAGE}
-    required_agent_capabilities: {CAPABILITIES}
-```
+<a name="adding-tests-to-ci"/>
 
-Where:
+There are several PRs needed to add a DeviceLab task to CI.
 
-- `{NAME_OF_TEST}` is the name of your test that also matches the name of the
+_TASK_- the name of your test that also matches the name of the
   file in `bin/tasks` without the `.dart` extension.
-- `{DESCRIPTION}` is the plain English description of your test that helps
-  others understand what this test is testing.
-- `{STAGE}` is `devicelab` if you want to run on Android, or `devicelab_ios` if
-  you want to run on iOS.
-- `{CAPABILITIES}` is an array that lists the capabilities required of
-  the test agent (the computer that runs the test) to run your test. As of writing,
-  the available capabilities are: `linux`, `linux/android`, `linux-vm`,
-  `mac`, `mac/ios`, `mac/iphonexs`, `mac/ios32`, `mac-catalina/ios`,
-  `mac-catalina/android`, `ios/gl-render-image`, `windows`, `windows/android`.
+
+1. Add prod builder to [flutter/infra devicelab_config.star](https://github.com/flutter/infra/blob/master/config/devicelab_config.star)
+  - Example PR: https://github.com/flutter/infra/pull/401/files
+  - This will need to soak for 15 minutes after merged to propagate
+  - There are various lists for the different testbeds a test can run on
+2. Add task to [flutter/flutter prod_builders.json](https://github.com/flutter/flutter/blob/master/dev/prod_builders.json)
+  - Example PR: https://github.com/flutter/flutter/pull/79913/files
+  - Set `flaky: true` for validation before blocking tree
+3. After 10 green runs, update [flutter/flutter prod_builders.json](https://github.com/flutter/flutter/blob/master/dev/prod_builders.json) to `flaky:false`
 
 If your test needs to run on multiple operating systems, create a separate test
 for each operating system.
+
+## Adding tests to presubmit
+
+Flutter's DeviceLab does not currently have capacity to run tests against physical devices in presubmit.
+
+Host only tests are okay. First add them to [continuous-integration](#adding-tests-to-ci), and after marked blocking add them to presubmit.
```

Comment:
Whoa, I didn't know this, neat.

In general I think we want to discourage these kinds of tests except in rare situations.  See https://github.com/flutter/flutter/issues/65790.

Maybe we can encourage people to consider adding a flutter_tools test instead?

Reply:
If that's the approach, i'll give the advice of adding them as flutter_tools tests. Is there documentation I should refer to?

Reply:
No, I think there are good use cases for adding host tests the devicelab, (for example the devicelab machines have the Apple development provisioning profiles set up, so https://github.com/flutter/flutter/pull/77428 for example can't been a flutter_tools test), and maybe some more desktop test reasons.

Maybe what you had before, but with a caveat like:
> Also consider creating your integration test in `packages/flutter_tools/test/integration.shard` to avoid using limited DeviceLab scheduling resources.  Example: https://github.com/flutter/flutter/pull/73577/files"

Reply:
Great! Added